lib/shopify_api/event_pipe/event_queue.ex

```diff
@@ -74,4 +74,18 @@ defmodule ShopifyAPI.EventPipe.EventQueue do
         Exq.subscribe(Exq, AuthToken.create_key(token), 1)
     end
   end
+
+  defp background_job_impl do
+    pid = GenServer.whereis(Exq)
+
+    if is_pid(pid) do
+      &Exq.enqueue/5
+    else
+      &run_inline/5
+    end
```

Comment:
I think using a `case` statement with a guard here would be more in keeping with the Elixir way of thinking. I find it a bit surprising to see `if` statements in Elixir.

Reply:
One of the reasons I prefer `case` statements in general is that they are easier to refactor into functions.

Reply:
going to leave this refactor for when the background job processor is configurable